src/include/uapi/ipc.h

```diff
@@ -404,46 +402,45 @@ struct sof_ipc_dai_config {
 #define SOF_IPC_MAX_CHANNELS			8
 
 /* channel positions - uses same values as ALSA */
-enum sof_ipc_chmap {
-	SOF_CHMAP_UNKNOWN = 0,
-	SOF_CHMAP_NA,		/* N/A, silent */
-	SOF_CHMAP_MONO,		/* mono stream */
-	SOF_CHMAP_FL,		/* front left */
-	SOF_CHMAP_FR,		/* front right */
-	SOF_CHMAP_RL,		/* rear left */
-	SOF_CHMAP_RR,		/* rear right */
-	SOF_CHMAP_FC,		/* front centre */
-	SOF_CHMAP_LFE,		/* LFE */
-	SOF_CHMAP_SL,		/* side left */
-	SOF_CHMAP_SR,		/* side right */
-	SOF_CHMAP_RC,		/* rear centre */
-	SOF_CHMAP_FLC,		/* front left centre */
-	SOF_CHMAP_FRC,		/* front right centre */
-	SOF_CHMAP_RLC,		/* rear left centre */
-	SOF_CHMAP_RRC,		/* rear right centre */
-	SOF_CHMAP_FLW,		/* front left wide */
-	SOF_CHMAP_FRW,		/* front right wide */
-	SOF_CHMAP_FLH,		/* front left high */
-	SOF_CHMAP_FCH,		/* front centre high */
-	SOF_CHMAP_FRH,		/* front right high */
-	SOF_CHMAP_TC,		/* top centre */
-	SOF_CHMAP_TFL,		/* top front left */
-	SOF_CHMAP_TFR,		/* top front right */
-	SOF_CHMAP_TFC,		/* top front centre */
-	SOF_CHMAP_TRL,		/* top rear left */
-	SOF_CHMAP_TRR,		/* top rear right */
-	SOF_CHMAP_TRC,		/* top rear centre */
-	SOF_CHMAP_TFLC,		/* top front left centre */
-	SOF_CHMAP_TFRC,		/* top front right centre */
-	SOF_CHMAP_TSL,		/* top side left */
-	SOF_CHMAP_TSR,		/* top side right */
-	SOF_CHMAP_LLFE,		/* left LFE */
-	SOF_CHMAP_RLFE,		/* right LFE */
-	SOF_CHMAP_BC,		/* bottom centre */
-	SOF_CHMAP_BLC,		/* bottom left centre */
-	SOF_CHMAP_BRC,		/* bottom right centre */
-	SOF_CHMAP_LAST = SOF_CHMAP_BRC,
-};
+#define SOF_CHMAP_UNKNOWN	0
```

Comment:
I think it's fine to use the enums here, it's easier to read.
The worry is only their representation for the IPC. And within the firmware itself the compiler can use the relevant representation

Reply:
You've lost me - can you give me an example of whet you mean here ?

Reply:
keep the enums instead of the defines, but don't use the type for the IPC. Leave the kernel and firmware use the enums elsewhere if they feel like it.

Reply:
You mean use uint32_t in the IPC cast to enum values ?